### PR TITLE
NetSim: Animate manually setting and reading wire

### PR DESCRIPTION
**Pull request by islemaster**
The visualization should animate the manual set/read processes in a way that helps the student understand what's going on - they are not "sending" bits, but "setting" a persistent state on the shared wire and "reading" that state as well.

Here's my first pass at that:
out

When "setting" the wire the new bit flies out from the node to the center of the wire, then fades away to a question mark indicating the "unknown" wire state.  When "reading" the wire the reverse happens - the wire state surfaces at its center, then flies to the local node.

The wire states are color-coded (Off/0/A = black, On/1/B = orange) and the colors will be easy to update from the stylesheet.

If "A and B" encoding is enabled and "Binary" encoding is not enabled, the bits in the visualization will be "A" and "B".  In any other configuration they will be "0" and "1"

### CODE REVIEW

File: apps/src/netsim/NetSimVizWire.js [hunk 1]
```diff
@@ -82,3 +161,153 @@ NetSimVizWire.prototype.kill = function () {
...
+  };
+};
+
+/**
+ * @returns {{x:number, y:number}}
+ */
+NetSimVizWire.prototype.getWireCenterPosition = function () {
+  return {
+    x: (this.remoteVizNode.posX - this.localVizNode.posX) / 2 +
+    this.localVizNode.posX,
```
Comment (Bjvanminnen) [inline apps/src/netsim/NetSimVizWire.js]: I would give this and line 311 an extra indent

File: apps/src/netsim/netsimUtils.js [hunk 2]
```diff
@@ -37,12 +38,72 @@ exports.jQuerySvgElement = function (type) {
...
+   */
+  newElement.toggleClass = function (className, shouldHaveClass) {
+    // Default second argument - if not provided, we flip the current state
+    shouldHaveClass = utils.valueOr(shouldHaveClass, !newElement.hasClass(className));
+
+    if (shouldHaveClass) {
+      newElement.addClass(className);
+    } else {
+      newElement.removeClass(className);
+    }
```
Comment (Bjvanminnen) [inline apps/src/netsim/netsimUtils.js]: This all looks okay to me, but I wonder if there's an existing jQuery plugin we could use that has run into/fixed any cross browser issues.

**Comment by Bjvanminnen** [comment]
lgtm. I think netsim is our cleanest looking app from a code perspective :)

File: apps/src/netsim/NetSimVizWire.js (hunk 1 above)
Comment (islemaster) [inline apps/src/netsim/NetSimVizWire.js]: I hate it when my formatting gets destroyed on copy/paste :(

**Comment by islemaster** [comment]
:boom: :astonished: :skull:  I find that hard to believe.

Thanks for the review!

File: apps/src/netsim/netsimUtils.js (hunk 2 above)
Comment (islemaster) [inline apps/src/netsim/netsimUtils.js]: There's [this one](http://keith-wood.name/svg.html) which is recently updated but it's not clear that it's extensively used.  The pickings are slim at the [jQuery plugin repository](https://plugins.jquery.com/tag/svg/).  This ["superdom plugin"](https://github.com/metaist/jquery.superdom) looks somewhat promising.

**Pull request merged**